docs/contributing.md

```diff
@@ -2,37 +2,92 @@
+
+* For changes that affect the *layer itself* such as adding new recipes or
+global functionality, please use `meta-raspberrypi` as the component name.
+* For changes that affect *layer recipes*, please just use the **base name**
+of the recipe as the component name. For example, use `omxplayer` instead of
+`omxplayer_git.bb`. This also applies when adding extra files to recipes.
+* For changes that affect the *layer documentation*, please just use `docs`
+as the component name.
+* For changes that affect *other files*, i.e. under the `conf` directory,
+please use the full path as the component name, e.g. `conf/layer.conf`.
 
-## Patches and pull requests
+A full example of a suitable commit log message is below:
 
-All the contributions should be compliant with the openembedded patch
-guidelines: <http://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines>
+    omxplayer: Update and add fixes for OpenSSL 1.1.x and thumb2
 
-To contribute to this project you should send pull requests to the github mirror
-(<https://github.com/agherzan/meta-raspberrypi>). **Additionally** you can send
-the patches for review to the above specified mailing list.
+    * Update to latest from upstream
+    * Backport OpenSSL 1.1.x related patches for ffmpeg
+    * Pass --cpu based on machine, instead of hardcoding it to arm1176jzf-s
+    * Make --cc, --cxx and --ld be used from environment, helps building
+      with clang
+    * It requires userland graphics driver to build to encode that in
+      COMPATIBLE_HOST
 
-When creating patches for the mailing list, please use something like:
+    Signed-off-by: Your Name <yourname@youremail.com>
+
+A common issue during patch reviewing is patch formatting. Please review
+the above formatting guidelines carefully before sending your patches.
+
+## Sending patches
+
+The preferred method to contribute to this project is to send pull
+requests to the GitHub mirror of the layer:
+
+* <https://github.com/agherzan/meta-raspberrypi>
+
+**In addition**, you may send patches for review to the above specified
+mailing list. In this case, when creating patches using `git` please make
+sure to use the following formatting for the message subject:
 
     git format-patch -s --subject-prefix='meta-raspberrypi][PATCH' origin
 
-When sending patches to the mailing list, please use something like:
+Then, for sending patches to the mailing list, you may use this command:
 
     git send-email --to yocto@yoctoproject.org <generated patch>
 
-## Github issues
+## GitHub issues
+
+In order to manage and track the layer issues more efficiently, the
+GitHub issues facility is used by this project:
+
+* <https://github.com/agherzan/meta-raspberrypi/issues>
+
+If you push patches that have a GitHub issue associated, please make sure to
+provide the corresponding issue number in the commit log message, just before
+the "Signed-off-by" line(s). For example:
+
+    omxplayer: Forward port patches and fix build
+
+    [Issue #324]
```

Comment:
There are github keywords which can change issue states etc. e.g. ```Fixed Issue #324``` etc will reflect correctly.

Reply:
I just used the original tag in the documentation, but yes I agree we should be using the standard GitHub keywords. Will reflect that. Thanks!